src/plugin/indexer-elastic-rest/src/java/org/apache/nutch/indexwriter/elasticrest/ElasticRestIndexWriter.java

```diff
@@ -0,0 +1,336 @@
+    private String host = null;
+    private String user = null;
+    private Boolean https = null;
+    private String password = null;
+    private Boolean trustAllHostnames = null;
+
+    private int maxBulkDocs;
+    private int maxBulkLength;
+    private long indexedDocs = 0;
+    private int bulkDocs = 0;
+    private int bulkLength = 0;
+    private boolean createNewBulk = false;
+    private long millis;
+    private BasicFuture<JestResult> basicFuture = null;
+
+    @Override
+    public void open(JobConf job, String name) throws IOException {
+
+        host = job.get(ElasticRestConstants.HOST);
+        port = job.getInt(ElasticRestConstants.PORT, 9200);
+        user = job.get(ElasticRestConstants.USER);
+        password = job.get(ElasticRestConstants.PASSWORD);
+        https = job.getBoolean(ElasticRestConstants.HTTPS, false);
+        trustAllHostnames = job.getBoolean(ElasticRestConstants.HOSTNAME_TRUST, false);
+
+        // trust ALL certificates
+        SSLContext sslContext = null;
+        try {
+            sslContext = new SSLContextBuilder().loadTrustMaterial(new TrustStrategy() {
+                public boolean isTrusted(X509Certificate[] arg0, String arg1) throws CertificateException {
+                    return true;
+                }
+            }).build();
+        } catch (NoSuchAlgorithmException | KeyManagementException | KeyStoreException e) {
+            e.printStackTrace();
+        }
+
+        // skip hostname checks
+        HostnameVerifier hostnameVerifier = null;
+        if (trustAllHostnames) {
+            hostnameVerifier = new NoopHostnameVerifier();
+        } else {
+            hostnameVerifier = new DefaultHostnameVerifier();
+        }
+
+        SSLConnectionSocketFactory sslSocketFactory = new SSLConnectionSocketFactory(sslContext);
+        SchemeIOSessionStrategy httpsIOSessionStrategy = new SSLIOSessionStrategy(sslContext, hostnameVerifier);
+
+        JestClientFactory jestClientFactory = new JestClientFactory();
+        URL urlOfElasticsearchNode = new URL(https ? "https" : "http", host, port, "");
+
+
+        if (host != null && port > 1) {
+            HttpClientConfig.Builder builder = new HttpClientConfig.Builder(urlOfElasticsearchNode.toString())
+                    .multiThreaded(true)
+                    .connTimeout(300000)
+                    .readTimeout(300000);
+            if (https) {
+                if (user != null && password != null) {
+                    builder.defaultCredentials(user, password);
+                }
+                builder.defaultSchemeForDiscoveredNodes("https")
+                        .sslSocketFactory(sslSocketFactory) // this only affects sync calls
+                        .httpsIOSessionStrategy(httpsIOSessionStrategy); // this only affects async calls
+
+            }
+            jestClientFactory.setHttpClientConfig(builder.build());
+        } else {
+            throw new RuntimeException("No host and port specified");
```

Comment:
Is there a more specific exception here over and above merely RuntimeException? It would be nice to LOG and capture it such that it is maintained within application logs. Thanks